Web Quiz Engine (Kotlin)/task/src/engine/WebQuizApiController.kt

```diff
@@ -24,6 +40,10 @@ class WebQuizApiController(private val quizService: QuizService) {
     fun addQuiz(@Valid @RequestBody quizDto: QuizDto): Quiz =
         quizService.addQuiz( quizDto = quizDto)
 
+    @DeleteMapping("/api/quizzes/{id}")
```

Comment:
Should return Http Status 204 and return type `Unit` or no return type.
See related comment in `QuizService.kt`.

Reply:
Now I fully understand https://github.com/f8full/hyperskill-web-quiz-engine/pull/4#discussion_r1577178264
You wouldn't want to see what I wrote before reading your present comment 😬

Reply:
I'm actually having trouble implementing this. When I do it as follow
```kotlin
@DeleteMapping("/api/quizzes/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteQuizById(@PathVariable(name = "id") quizId: Int) =
        quizService.deleteQuizById(quizId)
```
I get the following test error
```
java.lang.AssertionError: Wrong answer in test #159

DELETE /api/quizzes/2 should respond with status code 403, responded: 204
```

That seems legit as it's like the raised `QuizAuthorMismatchException` is not properly handled.
I am not sure how to fix it and will need your guidance on that one as well.

I did commit the code I just showed that fails test